### PR TITLE
docs: 📚 Adjust all demo projects to use the Synergy Form Field Demo

### LINKED ISSUE #142: docs: 📚 Adjust react demo project to use the Synergy Form Field Demo
# Description

After the recipe for using forms in vanilla HTML/JS #293 is implemented, the React Demo (located at `packages/_private/react-demo`) should be updated to mirror the functionality of the showcase. This includes Validation, form submission etc.

# Rationale

As a developer using React, I would like to have an example that shows how to build Forms using Synergy so I compare the results from #293 with my framework of choice next to another.

### LINKED ISSUE #143: docs: 📚 Adjust angular demo project to use the Synergy Form Field Demo
# Description

After the recipe for using forms in vanilla HTML/JS #293 is implemented, the Angular Demo (located at `packages/_private/angular-demo`) should be updated to mirror the functionality of the showcase. This includes Validation, form submission etc.

# Rationale

As a developer using Angular, I would like to have an example that shows how to build Forms using Synergy so I compare the results from #293 with my framework of choice next to another.

### LINKED ISSUE #209: docs: 📚 Adjust vue demo project to use the Synergy Form Field Demo
# Description

After the recipe for using forms in vanilla HTML/JS #293 is implemented, the Vue Demo (located at `packages/_private/vue-demo`) should be updated to mirror the functionality of the showcase. This includes Validation, form submission etc.

# Rationale

As a developer using Vue.js, I would like to have an example that shows how to build Forms using Synergy so I compare the results from #293 with my framework of choice next to another.

### PR DESCRIPTION
**Pull request by schilchSICKAG**
# Pull Request

## 📖 Description

This PR updates all framework wrappers to use the new vanilla demo created in #293. This makes sure we are able to write common automated tests in #260.

### 🎫 Issues

- Closes #142
- Closes #143 
- Closes #209

## 👩‍💻 Reviewer Notes

- All demos should look the same, as the same basic css file was used. Please note any differences
- Each element has an id attribute. It can be used for tests later on.
- Reset method is currently not possible in the vue demo because of #300. This issue will have to be solved first before reset and prefilled fields are working.
- Would also love to hear some feedback about the angualr solution from @kirchsuSICKAG ;)

## 📑 Test Plan

- Check out the branch and make sure it is up to date (`pnpm i -r && pnpm build`)
- Move to `packages/_private` and have a look at all the demos (`angular-demo`, `react-demo`, `vanilla-demo`, `vue-demo`).

## ✅ DoD

- [x] I have tested my changes manually.
- [ ] ~~I have added automatic tests for my changes (unit- and visual regression tests).~~
- [ ] ~~I have updated the project documentation to reflect my changes (e.g. CHANGELOG, Storybook, ...).~~
- [x] I have made sure to follow the projects coding and contribution guides.
- [ ] ~~I have made sure that the bundle size has changed appropriately.~~
- [ ] ~~I have validated that there are no accessibility errors.~~
- [x] I have used design tokens instead of fix css values

### CODE REVIEW

File: packages/_private/angular-demo/src/demoform/demoform.component.html [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+        <syn-option value="lead">Team Lead</syn-option>
+        <syn-option value="other">Other (please specify in comment section below)</syn-option>
+      </syn-optgroup>
+    </syn-select>
+
+    <syn-input
+      formControlName="name"
+      id="input-text"
+      label="Name"
+      [minlength]="10"
```
Comment (klaarseSICKAG) [inline packages/_private/angular-demo/src/demoform/demoform.component.html:45]: should be 5 if it should match to vanilla demo

File: packages/_private/angular-demo/src/app.component.html [hunk 2]
```diff
@@ -0,0 +1,8 @@
+<header>
+  <h1>synergy-design-system/vue Form Demo</h1>
```
Comment (klaarseSICKAG) [inline packages/_private/angular-demo/src/app.component.html:2]: NG demo

File: packages/_private/angular-demo/src/demoform/demoform.component.html [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+      id="checkbox-newsletter-beta"
+      name="newsletterBeta"
+    >
+      I am interested in the Synergy Beta Program
+    </syn-switch>
+  </demo-fieldset>
+  <!-- /Marketing -->
+
+  <syn-divider />
+
```
Comment (klaarseSICKAG) [inline packages/_private/angular-demo/src/demoform/demoform.component.html:189]: Additional information section is missing

File: packages/_private/vue-demo/src/DemoForm.vue [hunk 4]
```diff
@@ -0,0 +1,323 @@
...
+        name="newsletterBeta"
+        v-model="formData.newsletterBeta"
+      >
+        I am interested in the Synergy Beta Program
+      </SynVueSwitch>
+    </DemoFieldset>
+    <!-- /Marketing -->
+
+    <SynVueDivider />
+
```
Comment (klaarseSICKAG) [inline packages/_private/vue-demo/src/DemoForm.vue:282]: Additional Information section is missing

File: packages/_private/vue-demo/src/DemoForm.vue [hunk 5]
```diff
@@ -0,0 +1,323 @@
...
+        <SynVueOptgroup label="Other">
+          <SynVueOption value="lead">Team Lead</SynVueOption>
+          <SynVueOption value="other">Other (please specify in comment section below)</SynVueOption>
+        </SynVueOptgroup>
+      </SynVueSelect>
+
+      <SynVueInput
+        id="input-text"
+        label="Name"
+        :minlength="10"
```
Comment (klaarseSICKAG) [inline packages/_private/vue-demo/src/DemoForm.vue:136]: 5?

File: packages/_private/react-demo/src/DemoForm.tsx [hunk 6]
```diff
@@ -0,0 +1,291 @@
...
+        <SynSwitch
+          checked={formData.newsletterBeta}
+          id="checkbox-newsletter-beta"
+          name="newsletterBeta"
+        >
+          I am interested in the Synergy Beta Program
+        </SynSwitch>
+      </DemoFieldset>
+      {/* /Marketing */}
+
```
Comment (klaarseSICKAG) [inline packages/_private/react-demo/src/DemoForm.tsx:279]: Additional Information section missing

File: packages/_private/angular-demo/src/demoform/demoform.component.html (hunk 1 above)
Comment (schilchSICKAG) [reply packages/_private/angular-demo/src/demoform/demoform.component.html]: I initially had problems getting this to work at all using TypeScript. Adjusted :)

File: packages/_private/angular-demo/src/app.component.html (hunk 2 above)
Comment (schilchSICKAG) [reply packages/_private/angular-demo/src/app.component.html]: Fixed

File: packages/_private/angular-demo/src/demoform/demoform.component.html (hunk 3 above)
Comment (schilchSICKAG) [reply packages/_private/angular-demo/src/demoform/demoform.component.html:189]: How did I miss that?

File: packages/_private/vue-demo/src/DemoForm.vue (hunk 4 above)
Comment (schilchSICKAG) [reply packages/_private/vue-demo/src/DemoForm.vue:284]: How did I miss that?

File: packages/_private/vue-demo/src/DemoForm.vue (hunk 5 above)
Comment (schilchSICKAG) [reply packages/_private/vue-demo/src/DemoForm.vue]: I initially had problems getting this to work at all using TypeScript. Adjusted :)

**Review by klaarseSICKAG — APPROVED** [review]

File: packages/_private/vue-demo/src/DemoForm.vue [hunk 7]
```diff
@@ -0,0 +1,339 @@
...
+  e.preventDefault();
+  e.stopPropagation();
+
+  const formElement = e.target as HTMLFormElement;
+  const isValid = formElement.checkValidity();
+
+  if (isValid) {
+    // eslint-disable-next-line no-alert
+    alert('Your data was successfully submitted');
+  }
```
Comment (kirchsuSICKAG) [inline packages/_private/vue-demo/src/DemoForm.vue:70]: We should make the validation handling the same for each wrapper. in *angular* and here in *vue* we only do an alert if its valid, in *vanilla* and *react* we do following:
image

File: packages/_private/vue-demo/src/DemoForm.vue [hunk 8]
```diff
@@ -0,0 +1,339 @@
...
+        required
+        v-model="formData.phone"
+        type="tel"
+      />
+
+      <SynVueInput
+        id="input-date"
+        label="Date of birth"
+        name="date"
+        placeholder="Please insert your E-mail address"
```
Comment (kirchsuSICKAG) [inline packages/_private/vue-demo/src/DemoForm.vue:171]: Should be removed. Is the message from email and a placeholder is not needed for date anyway. This needs to be fixed in all wrappers

File: packages/_private/vanilla-demo/index.html
Comment (kirchsuSICKAG) [inline packages/_private/vanilla-demo/index.html:1]: The placeholder on the  `input-date` syn-input Should be removed. It's the message from email and a placeholder is not needed for date anyway. This needs to be fixed in all wrappers

File: packages/_private/react-demo/src/DemoForm.tsx [hunk 9]
```diff
@@ -0,0 +1,309 @@
...
+          required
+          value={formData.phone}
+          type="tel"
+        />
+
+        <SynInput
+          id="input-date"
+          label="Date of birth"
+          name="date"
+          placeholder="Please insert your E-mail address"
```
Comment (kirchsuSICKAG) [inline packages/_private/react-demo/src/DemoForm.tsx:170]: Should be removed. Is the message from email and a placeholder is not needed for date anyway. This needs to be fixed in all wrappers

File: packages/_private/angular-demo/src/demoform/demoform.component.html [hunk 10]
```diff
@@ -0,0 +1,211 @@
...
+      [required]="true"
+      type="tel"
+    />
+
+    <syn-input
+      formControlName="date"
+      id="input-date"
+      label="Date of birth"
+      name="date"
+      placeholder="Please insert your E-mail address"
```
Comment (kirchsuSICKAG) [inline packages/_private/angular-demo/src/demoform/demoform.component.html:78]: Should be removed. Is the message from email and a placeholder is not needed for date anyway. This needs to be fixed in all wrappers

**Review by kirchsuSICKAG — CHANGES_REQUESTED** [review]

**Pull request merged**